src/finishRes.ts

```diff
@@ -42,10 +43,37 @@ export function finishRes(res: ValidatableResponse, status: number, json: Format
   if (isErrorsResponse(json)) {
     json.errors.forEach(error => {
       if (error instanceof Error) {
+
+        // `RippledError`s contain `error` (e.g. `actNotFound` when source account is not found)
+        const name = ((error as any).data && (error as any).data.error) || error.name;
+
+        // `RippledError`s contain `error_code` (e.g. `19` for `actNotFound`)
+        let code = (error as any).code || ((error as any).data && (error as any).data.error_code) || ERRORS.CODES.UNSPECIFIED;
+
+        if (name === 'DisconnectedError') {
+          // name: DisconnectedError
+          // message: websocket was closed
+          code = ERRORS.CODES.WEBSOCKET;
+          if (!json.message) {
+            json.message = 'Lost connection to rippled. Please try again.';
+          }
+        }
+
+        if (name === 'actNotFound') {
+          const data = (error as any).data;
+          if (!json.message && data.account && data.ledger_current_index && data.request.command) {
+            json.message = `The account (${data.account}) could not be found as of ledger ${data.ledger_current_index} (command: ${data.request.command})`;
+          }
+        }
+
+        if (!json.message) {
+          json.message = `${name}: ${error.message}`;
+        }
```

Comment:
Can you pull this "setting the default json.message value" logic out into a separate pass? It would seperate the two and allow you to avoid modifying the state of the json object mid-loop: 

```
const serializedErrors = json.errors.map(...
json.message = json.message || getMessageFromSerializedErrors(serializedErrors)
```